pkg/docker/pipeline_push.go

```diff
@@ -210,7 +211,19 @@ func validateRequirements(projectDir string, client *http.Client, cfg *config.Co
 				return err
 			}
 			if pipelinePackage == projectPackage {
-				found = pipelineVersion == "" || version.GreaterOrEqual(projectVersion, pipelineVersion)
+				if pipelineVersion == "" || projectVersion == pipelineVersion {
+					found = true
+				} else {
+					leftVersion, err := version.Parse(projectVersion)
+					if err != nil {
+						return err
+					}
+					rightVersion, err := version.Parse(pipelineVersion)
+					if err != nil {
+						return err
+					}
+					found = leftVersion.GreaterThanOrEqual(rightVersion)
+				}
 				break
```

Comment:
I think what we want to do here is parse the pipelinePackage and ensure that the range provided by the project requirements.txt is satisfied. From the docs this looks like:

```go
v, _ := version.Parse("2.1")
c, _ := version.NewSpecifiers(">= 1.0, < 1.4 || > 2.0")

if c.Check(v) {
	fmt.Printf("%s satisfies specifiers '%s'", v, c)
}
```

Reply:
Resolved in [89d5f26](https://github.com/replicate/cog/pull/2415/commits/89d5f263e10f79f8fc3e2ee1c78a8ee27f7fe4ab)